core/src/main/java/io/micronaut/core/type/Argument.java

```diff
@@ -493,6 +493,19 @@ static <T> Argument<T> of(
         return new DefaultArgument<>(type, null, AnnotationMetadata.EMPTY_METADATA, Collections.emptyMap(), Argument.ZERO_ARGUMENTS);
     }
 
+    /**
+     * Creates a new argument for the type of the given instance.
+     *
+     * @param instance The argument instance
+     * @param <T>      The generic type
+     * @return The argument instance
+     * @since 4.6
+     */
+    @NonNull
+    static <T> Argument<T> of(T instance) {
```

Comment:
I don't think it's a good idea to overload `of` for this, it'll just lead to accidental calls

Reply:
I can rename the method. Otherwise I will just inline

Reply:
Either is fine with me